src/pages/workspace/expensifyCard/WorkspaceEditCardLimitPage.tsx

```diff
@@ -0,0 +1,154 @@
+import type {FormInputErrors, FormOnyxValues} from '@components/Form/types';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import ScreenWrapper from '@components/ScreenWrapper';
+import useAutoFocusInput from '@hooks/useAutoFocusInput';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as CurrencyUtils from '@libs/CurrencyUtils';
+import * as ValidationUtils from '@libs/ValidationUtils';
+import Navigation from '@navigation/Navigation';
+import type {SettingsNavigatorParamList} from '@navigation/types';
+import AccessOrNotFoundWrapper from '@pages/workspace/AccessOrNotFoundWrapper';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import INPUT_IDS from '@src/types/form/EditExpensifyCardLimitForm';
+
+type ConfirmationWarningTranslationPaths = 'workspace.expensifyCard.smartLimitWarning' | 'workspace.expensifyCard.monthlyLimitWarning' | 'workspace.expensifyCard.fixedLimitWarning';
+
+// TODO: remove when Onyx data is available
+const mockedCard = {
+    accountID: 885646,
+    availableSpend: 1000,
+    nameValuePairs: {
+        cardTitle: 'Test 1',
+        isVirtual: true,
+        limit: 2000,
+        limitType: CONST.EXPENSIFY_CARD.LIMIT_TYPES.SMART,
+    },
+    lastFourPAN: '1234',
+};
+
+type WorkspaceEditCardLimitPageProps = StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.EXPENSIFY_CARD_LIMIT>;
+
+function WorkspaceEditCardLimitPage({route}: WorkspaceEditCardLimitPageProps) {
+    const {policyID, cardID} = route.params;
+    const {translate} = useLocalize();
+    const {inputCallbackRef} = useAutoFocusInput();
+    const styles = useThemeStyles();
+    const [isConfirmModalVisible, setIsConfirmModalVisible] = useState(false);
+
+    const [cardsList] = useOnyx(`${ONYXKEYS.COLLECTION.WORKSPACE_CARDS_LIST}${policyID}_${CONST.EXPENSIFY_CARD.BANK}`);
+    const card = cardsList?.[cardID] ?? mockedCard;
+
+    const getPromptTextKey = useMemo((): ConfirmationWarningTranslationPaths => {
+        switch (card.nameValuePairs?.limitType) {
+            case CONST.EXPENSIFY_CARD.LIMIT_TYPES.SMART:
+                return 'workspace.expensifyCard.smartLimitWarning';
+            case CONST.EXPENSIFY_CARD.LIMIT_TYPES.FIXED:
+                return 'workspace.expensifyCard.fixedLimitWarning';
+            case CONST.EXPENSIFY_CARD.LIMIT_TYPES.MONTHLY:
+                return 'workspace.expensifyCard.monthlyLimitWarning';
+            default:
+                return 'workspace.expensifyCard.fixedLimitWarning';
+        }
+    }, [card.nameValuePairs?.limitType]);
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const updateCardLimit = (newLimit: string) => {
+        setIsConfirmModalVisible(false);
+
+        // TODO: add API call when it's supported https://github.com/Expensify/Expensify/issues/407831
```

Comment:
The API endpoint `UpdateExpensifyCardLimit` is live on staging, can we integrate it as part of this PR? :)

Reply:
Sure, I will fo it!

Reply:
@MariaHCD Is it possible for me to test this endpoint over the realistic card in the E/App?

Reply:
Ah, no, not yet. The backend PR to configure a workspace for the Expensify Card is still WIP so we can't test any issuing cards or updating cards at this point.

We can do full E2E tests once more of the backend pieces are implemented.

Reply:
@MariaHCD I've made a call for non-existing card and it returned `200` response. Is it okay?
I expected something with `404` 😅 
![image](https://github.com/user-attachments/assets/bb1cfdd6-3bc8-42a0-be63-78e49bf92366)

Reply:
Integrated `UpdateExpensifyCardLimit` API call: https://github.com/Expensify/App/pull/45397/commits/5e311257e5b683b2f3cab2f64bce3c8050a30ae7

Reply:
Ah, thanks for catching that. Looking at our [backend logs](https://www.expensify.com/_devportal/tools/logSearch/#sort=asc&size=10000&query=request_id%3A%228a5303eac9bec04a-WAW%22%20AND%20timestamp%3A%5B2024-07-18T00%3A00%20TO%202024-07-18T23%3A59%5D), we did throw a `404` but the API didn't rethrow the exception. cc: @nkuoch